xrpackage.js

```diff
@@ -1,1763 +1,4 @@
-
-    const d = p.getMainData();
-    const indexHtml = d.toString('utf8');
-    await p.context.iframe.contentWindow.xrpackage.iframeInit({
-      engine: this,
-      pkg: p,
-      indexHtml,
-      context: this.proxyContext,
-      id: p.id,
-      schema: p.schema,
-      xrState: this.xrState,
-      XRPackage,
-    });
-    const xrfb = this.realSession ? this.realSession.renderState.baseLayer.framebuffer : this.fakeXrFramebuffer;
-    p.setXrFramebuffer(xrfb);
-
-    await p.context.requestPresentPromise;
-  },
-  'gltf@0.0.1': async function(p) {
-    this.container.add(p.context.object);
-  },
-  'vrm@0.0.1': async function(p) {
-    this.container.add(p.context.object);
-  },
-  'vox@0.0.1': async function(p) {
-    this.container.add(p.context.object);
-  },
-};
-const xrTypeRemovers = {
-  'webxr-site@0.0.1': function(p) {
-    this.rafs = this.rafs.filter(raf => {
-      const rafWindow = raf[symbols.windowSymbol];
-      const rafPackage = this.packages.find(p => p.context.iframe && p.context.iframe.contentWindow === rafWindow);
-      return rafPackage !== p;
-    });
-
-    const emitKeyboardEvent = p.context.emitKeyboardEvent;
-
-    window.removeEventListener('keydown', emitKeyboardEvent, true);
-    window.removeEventListener('keyup', emitKeyboardEvent, true);
-    window.removeEventListener('keypress', emitKeyboardEvent, true);
-
-    p.context.iframe && p.context.iframe.parentNode.removeChild(p.context.iframe);
-  },
-  'gltf@0.0.1': function(p) {
-    this.container.remove(p.context.object);
-  },
-  'vrm@0.0.1': function(p) {
-    this.container.remove(p.context.object);
-  },
-  'vox@0.0.1': function(p) {
-    this.container.remove(p.context.object);
-  },
-};
-
-const _setFramebufferMsRenderbuffer = (gl, xrfb, width, height, devicePixelRatio) => {
-  const oldDrawFbo = gl.getParameter(gl.DRAW_FRAMEBUFFER_BINDING);
-  const oldRbo = gl.getParameter(gl.RENDERBUFFER_BINDING);
-  
-  const colorRenderbuffer = gl.createRenderbuffer();
-  const depthRenderbuffer = gl.createRenderbuffer();
-  const colorTex = gl.createTexture();
-  const depthTex = gl.createTexture();
-
-  gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, xrfb);
-
-  gl.bindRenderbuffer(gl.RENDERBUFFER, colorRenderbuffer);
-  gl.renderbufferStorageMultisample(gl.RENDERBUFFER, 4, gl.RGBA8, width * devicePixelRatio, height * devicePixelRatio);
-  gl.framebufferRenderbuffer(gl.DRAW_FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.RENDERBUFFER, colorRenderbuffer);
-
-  gl.bindRenderbuffer(gl.RENDERBUFFER, depthRenderbuffer);
-  gl.renderbufferStorageMultisample(gl.RENDERBUFFER, 4, gl.DEPTH32F_STENCIL8, width * devicePixelRatio, height * devicePixelRatio);
-  gl.framebufferRenderbuffer(gl.DRAW_FRAMEBUFFER, gl.DEPTH_STENCIL_ATTACHMENT, gl.RENDERBUFFER, depthRenderbuffer);
-  
-  gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, oldDrawFbo);
-  gl.bindRenderbuffer(gl.RENDERBUFFER, oldRbo);
-};
-export class XRPackageEngine extends EventTarget {
```

Comment:
I think to make sure `xrpackage.js` still works with everything that uses it, we need to keep `export`ing the exports from the file like this one, which looks like it's just `XRPackageEngine`, `XRPackage`, and `apiHost`

Reply:
something like `export * from './xrpackage-api.js';` might do I think